tls/s2n_certificate_keys.c

```diff
@@ -82,3 +82,29 @@ const struct s2n_certificate_key s2n_ec_p521 = {
     .name = "ecdsa_p521",
     .bits = 521,
 };
+
+const struct s2n_certificate_key *s2n_certificate_keys_rfc9151[] = {
+    /**
+     *= https://www.rfc-editor.org/rfc/rfc9151#section-5.1
+     *# CNSA (D)TLS connections MUST use secp384r1
+     **/
+    &s2n_ec_p384,
+
+    /**
+     *= https://www.rfc-editor.org/rfc/rfc9151#section-5.2
+     *# CNSA specifies a minimum modulus size of 3072 bits; however, only two
+     *# modulus sizes (3072 bits and 4096 bits) are supported by this profile.
+     **/
+    &s2n_rsa_rsae_3072,
+    &s2n_rsa_rsae_4096,
+
+    /* OID:rsassa-PSS keys are allowed by rfc9151, but s2n-tls does not currently
+     * support validating the digest size on RSA-PSS signatures, so they are not
+     * allowed in s2n_certificate_signature_preferences_rfc9151. \
+     */
```

Comment:
This seems like a big miss. Do we have an issue for it?

Right now, couldn't that cause this policy to reject valid certificates? Couldn't that break compliant customers?

Also, if the certificate_signature_preferences are the problem, should we put RSA-PSS here anyway?

Reply:
> Do we have an issue for it?

#3435 

> couldn't that cause this policy to reject valid certificates? Couldn't that break compliant customers?

Yes, and yes. I agree that we need to add support for it, and once we have, we can add rsa-pss signatures into the certificate preferences.

This does mean that some technically compliant customers might not be able to negotiate with this policy, which is unfortunate but understandable. I think our messaging should be that 
- any negotiation with `$COMPLIANCE` policy is compliant
- but we do not necessarily support all possible compliant connections

> if the certificate_signature_preferences are the problem, should we put RSA-PSS here anyway?

I would vote against it because it just feels inconsistent? And having it listed when we don't actually allow RSA-PSS signature is a "no-op".

Reply:
I'd still vote for adding RSA-PSS here. It's back to the "single source of truth" concern I have with the current setup: the problem is with the certificate_signature_preferences, so if I fix the certificiate_signature_preferences it should just start working. I shouldn't also need to remember to fix it in certificate_types.